### PR TITLE
Add Circle `as_rect()`/`as_frect()`

### PR DESCRIPTION
**Pull request by itzpr3d4t0r**
This PR adds two functions for turning a Circle object to a Rect/Frect object. 

## Credits

**Geometry Project:**
For code, docs and tests:
@novialriptide @Emc2356 @itzpr3d4t0r @ScriptLineStudios @avaxar @Matiiss @newpaxonian @maqa41 @blankRiot96
Also thanks to @Starbuck5 for kickstarting the idea and the occasional help!

**Functionality added in this PR**
- `circle.c`
https://github.com/pygame-community/pygame-geometry/blame/main/src_c/circle.c
Credits to @Emc2356 @itzpr3d4t0r @novialriptide
- `geometry.pyi`
https://github.com/pygame-community/pygame-geometry/blame/main/geometry.pyi
Credits to @Emc2356 @itzpr3d4t0r @novialriptide @ScriptLineStudios @avaxar 
- `geometry_test.py`
https://github.com/pygame-community/pygame-geometry/blame/main/test/test_circle.py
Credits to @itzpr3d4t0r @novialriptide
- `geometry.rst`
https://github.com/pygame-community/pygame-geometry/blame/main/docs/circle.rst
Credits to @itzpr3d4t0r

### CODE REVIEW

**Review by ankith26 — COMMENTED** [review]
While I don't hate the current name, should we consider alternatives? Something like `get_bounding_rect`?

**Comment by itzpr3d4t0r** [comment]
I mean i tried to go with something short but `get_bounding_rect` was what we named it for Polygon and then renamed it to as_rect. "get_bounding_rect" describes the method better tho.

File: docs/reST/ref/geometry.rst [hunk 1]
```diff
@@ -255,6 +255,40 @@
...
+         object returned by as_rect will have dimensions such that it completely encloses
+         the `Circle`, with no part of the `Circle` extending outside of the `FRect`.
+
+         .. note::
+             This method is equivalent(behaviour wise) to the following code:
+             ::
+                 FRect(circle.x - circle.r, circle.y - circle.r, circle.r * 2, circle.r * 2)
+
+         .. ## Circle.as_frect ##
+
```
Comment (ankith26) [inline docs/reST/ref/geometry.rst:291]: From now on, using versionadded tags may be a good idea. Both these functions need a versionadded 2.5.0 tag.

Actually now that I see this, the entire module is missing it. Maybe one note at the top of this doc saying the module was added in 2.4.0 should be added?

**Review by ankith26 — APPROVED** [review]
LGTM, thanks for the PR :tada: 

Left one review comment for your consideration

File: docs/reST/ref/geometry.rst (hunk 1 above)
Comment (itzpr3d4t0r) [reply docs/reST/ref/geometry.rst:291]: Thanks, I'll add the versionadded for these methods in this PR and all the others in a separate one.

**Review by MyreMylar — APPROVED** [review]
LGTM 👍

**Pull request merged**